docs/developer/feeds.rst

```diff
@@ -4,122 +4,199 @@
+    agent.register_feed(feed_name)
 
+The ``register_feed`` function takes a few other key word arguments to customize
+its behavior (See :ref:`Feed Api <ocs_feed_api>` for more details).
+``buffer_time`` will set how long the feed should buffer messages before sending
+over crossbar, and ``max_messages`` will set how many messages are cached.
 
+Feed_name Rules
+````````````````
 
-Subscribing to a Feed
----------------------
-
-To subscribe to a feed you can call::
-
-    agent.subscribe_to_feed(agent_addr, feed_name, callback, force_subscribe = False)
-
-You must pass the address of the agent that registered the feed, and the name of
-the feed along with a callback function that is called whenever the feed is
-published to.
-
-The callback is passed the tuple ``(message, feed_data)``
-where ``message`` is the published data, and ``feed_data`` is a dictionary
-with the fields seen above.
-An example of how this is used is seen in the Aggregator agent::
-
-    def add_feed(self, agent_address, feed_name):
-        """
-        Subscribes to aggregated feed
+Data on this feed will be published to the URI ``<agent_uri>.feeds.<feed_name>``
+so feed names must follow standard
+`URI formatting rules <https://crossbar.io/docs/URI-Format/>`_, meaning
+the feed can only use lowercase letters, numbers, and underscores.
+As the ``agent_uri`` will be unique for each agent, feed names are not required
+to be globally unique.
 
-        Args:
-            agent_address (string):
-                agent address of the feed.
+For instance, the Lakeshore372 agent may register a feed called ``temperatures``.
+A Lakeshore372 instance with instance-id ``LSASIM`` would then publish data to
+the URI ``observatory.LSASIM.feeds.temperatures``.
 
-            feed_name (string):
-                name of the feed.
-        """
+Agg Params
+```````````
+If you'd like your feed to be recorded by the hk aggregator, you must register
+with the keyword ``record=True``, and you can customize the hk aggregator
+behavior with the ``agg_params`` option.
+The following options can be specified here:
 
-        def _data_handler(_data):
-            """Callback whenever data is published to an aggregated feed"""
-            data, feed = _data
-            if self.aggregate:
-                self.incoming_data[feed["address"]].append(data)
+.. list-table:: Aggregator params
+    :widths: 20 20
 
-        feed_address = "{}.feeds.{}".format(agent_address, feed_name)
-        if feed_address in self.incoming_data.keys():
-            return
+    * - frame_length
+      - Aggregation time (seconds) before frame is written to disk
 
-        self.agent.subscribe_to_feed(agent_address, feed_name, _data_handler)
-        self.incoming_data[feed_address] = []
-        self.log.info("Subscribed to feed {}".format(feed_address))
+    * - fresh_time
+      - Time (seconds) before feed is considered "stale", and is removed from
+        the HK status frame
```

Comment:
Where are the agg_params actually decoded?  What are the defaults?  I assume this is in aggregator agent somewhere, and if so it would be good to link to it.

Also, this is now a superset of what is described in register_agent docstring.  So it's possible we've got this information in 3 different places now...

Reply:
This is an argument to the Provider, so they're described in the Provider docstring.